forum/qa-plugin/report-reason/q2apro-flag-reasons-admin.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+class q2apro_flagreasons_admin
+{
+    const NOTICE_LENGTH = 255;
+
+    public function init_queries($tablesLc)
+    {
+        $tableName = qa_db_add_table_prefix('flagreasons');
+        $result    = null;
+
+        if (!in_array($tableName, $tablesLc, true)) {
+            require_once QA_INCLUDE_DIR . 'qa-app-users.php';
+
+            $result = '
+                CREATE TABLE IF NOT EXISTS `' . $tableName . '` (
+                  `userid` int(10) UNSIGNED NOT NULL,
+                  `postid` int(10) UNSIGNED NOT NULL,
+                  `reasonid` int(10) UNSIGNED NOT NULL,
+                  `notice` varchar(self::NOTICE_LENGTH) NULL,
```

Comment:
zapewne to powoduje błąd z instalacją, próbuje w bazie stworzyć kolumnę typu varchar(self::NOTICE_LENGTH), nie ma tu podstawienia wartości z tej stałej

Reply:
Być może, to jest oryginalny kod backendowy z któregoś bazowego brancha. Jeśli trzeba to poprawić, to poproszę o wskazówki jak to zrobić.

Reply:
może, ale wcześniej instalacja działała, a to powoduje pokazany na początku głównego komentarza błąd

tutaj bo bazy idzie zapytanie, że ma dodać kolumnę `notice` varchar(self::NOTICE_LENGTH) NULL, a to nie jest możliwe, baza nie wie co to jest self::NOTICE_LENGTH, powinno być varchar(255) (czy tam inna długość). Wystarczy więc zrobić tu np. konkatenacje aby wstawić wartość tak samo jak chociażby przy nazwie tabeli i będzie ok

Reply:
Ah, przepraszam - faktycznie zmieniłem konfigurację kolumny "notice" na `varchar` o konkretnej długości ustalanej z property `NOTICE_LENGTH` (ta wartość idzie również do frontu, który na tej podstawie ustawia maksymalną długość <textarea> blokując wpisanie większej liczby znaków niż ta określona).